client/lib/shortcode/index.js

```diff
@@ -14,9 +14,8 @@ const REGEXP_SHORTCODE = /\[(\[?)([^[\]/\s\u00a0\u200b]+)(?![\w-])([^\]/]*(?:\/(
  * Numeric attributes can be formatted as `"value"` or just `value`.
  *
  * @see https://github.com/WordPress/WordPress/blob/HEAD/wp-includes/js/shortcode.js
- *
  * @param  {string} text A shortcode attribute string
- * @returns {object}      An object of attributes, split as named and numeric
+ * @returns {{named: Object.<string, string>, numeric: string[]}} An object of attributes, split as named and numeric
```

Comment:
is this supposed to be `Record<string, string>`?

Reply:
In JSDoc, you have to use `Object.<string, string>` as `Record` doesn't exist (the linter will even tell you this).